packages/utils/src/normalize.ts

```diff
@@ -98,17 +98,27 @@ function visit(
     return stringified;
   }
 
-  // We're also done if we've reached the max depth
-  if (depth === 0) {
-    // At this point we know `serialized` is a string of the form `"[object XXXX]"`. Clean it up so it's just `"[XXXX]"`.
-    return stringified.replace('object ', '');
-  }
+  // From here on, we can assert that `value` is either an object or an array!
 
   // If we've already visited this branch, bail out, as it's circular reference. If not, note that we're seeing it now.
   if (memoize(value)) {
     return '[Circular ~]';
   }
 
+  // Do not normalize objects that we know have already been normalized. This MUST be below the circ-ref check otherwise
+  // we might somehow accidentally produce circular references by skipping normalization.
+  // As a general rule, the "__sentry_skip_normalization__" property should only be used sparingly and only should only
+  // be set on objects that have already been normalized.
+  if ((value as ObjOrArray<unknown>)['__sentry_skip_normalization__']) {
+    return value as ObjOrArray<unknown>;
+  }
+
+  // We're also done if we've reached the max depth
+  if (depth === 0) {
+    // At this point we know `serialized` is a string of the form `"[object XXXX]"`. Clean it up so it's just `"[XXXX]"`.
+    return stringified.replace('object ', '');
```

Comment:
Why did we move this below the memoize?

Reply:
I thought there was a case where we need to move the `memoize` call to avoid circular references with objects for which we skip normalization. Your comment made me think about it again and

a) I think there is actually no such case
b) if there were such a case, moving it is pointless because skipping normalization is already "dangerous" in that regard.

Anyways, moved it back, thanks ^^ -> [77e3096](https://github.com/getsentry/sentry-javascript/pull/5052/commits/77e3096bbfca3c1af732b8ed00f98fb6b71f2a40)